Refactoring

SemPlaatsman opened the pull request.

SemPlaatsman committed:
Valid Role enum in User

SemPlaatsman committed:
First code without errors

- Split up all relevant functionality into auth and identity.
- Added new endpoints (will be converted to rpc in the future).
- Added proper keycloak entities.

SemPlaatsman committed:
Fixed README >:)

SemPlaatsman committed:
Works I guess (bruh moment)

SemPlaatsman committed:
Removed PIN from CreateIdentityRequest

RamonKea-inh commented on core/services/encryption_service.go:
```diff
@@ -0,0 +1,152 @@
+	"fmt"
+	"strings"
+
+	"golang.org/x/crypto/argon2"
```
Wat is x/crypto/argon2?

SemPlaatsman replied:
Argon2 is een encryption algorithme. `x/crypto/argon2` is de library die daarbij helpt.

RamonKea-inh commented on infrastructure/repositories/base_keycloak_repository.go:
```diff
@@ -0,0 +1,162 @@
+		return nil, fmt.Errorf("request failed: %w", err)
+	}
+
+	log.Debug("Received HTTP response",
```
Debug code?

SemPlaatsman replied:
Zijn debug logs. Die zijn dus alleen zichtbaar als je in debugging mode opstart

SemPlaatsman committed:
Valid request body casing

SemPlaatsman committed:
Fixed role update

SemPlaatsman committed:
Fixed role and password update functionality

SemPlaatsman committed:
Merge branch 'refactoring' of https://github.com/PTSS-Support/Authentication-Service into feature/identity-storage

SemPlaatsman committed:
Added create pin endpoint and fixed update pin

Tayam-El-Jebari commented on README.md:
````diff
@@ -28,22 +28,40 @@ server:
 ```
 
-5. Run the application
+5. Configure Keycloak
+   Follow the [Keycloak Setup Guide](docs/KEYCLOAK_SETUP_GUIDE.md) to:
+   - Create a custom realm
+   - Configure the required client
+   - Set up proper authentication flows
````
heads up, found a way to automate this, has a bug rn, but will automate this step entirely.

SemPlaatsman replied:
Woohoo, then this can be deleted later. Watcha using?

Tayam-El-Jebari replied:
docker compose will call a keycloak-init.sh script which will do these steps using curl commands, saves headache and will update a .env file with the values (can be changed to config.yaml if we want :) )

SemPlaatsman replied:
Nahhh we using .env from now on. It integrates with platform-config

Tayam-El-Jebari commented on README.md:
btw, general question, but why did you opt for a config.yaml instead of .env? Should work fine. Not necessary though to change this btw.

SemPlaatsman replied:
We need to use .env in the future, thought Go needed config.yaml because that's what was in the initial README. Will update soon then.

Tayam-El-Jebari commented on api/controllers/auth_controller.go:
```diff
@@ -2,14 +2,14 @@ package controllers
-	"strings"
 
-	"github.com/PTSS-Support/identity-service/api/dtos/requests"
+	requests "github.com/PTSS-Support/identity-service/api/dtos/requests/auth"
```
tested this in code, but why have requests here requests here at the beginning whilst all other classes don't use it?
Inconsistent smh
but my ide gave no errors when removing it, so why have it?

Tayam-El-Jebari replied:
nvm found out that you have it quite a lot in this code. Is this necessary though?

SemPlaatsman replied:
Idk why it doesn't work. My suspicion is that it has something to do with the folder path (e.g. namespace does not path last folder naming) but idk.

Tayam-El-Jebari replied:
hmm weird, alright, lets keep it then

Tayam-El-Jebari commented on core/facades/auth_facade.go:
```diff
@@ -3,15 +3,13 @@ package facades
 
-	"github.com/PTSS-Support/identity-service/api/dtos/requests"
-	"github.com/PTSS-Support/identity-service/api/dtos/responses"
+	requests "github.com/PTSS-Support/identity-service/api/dtos/requests/auth"
+	responses "github.com/PTSS-Support/identity-service/api/dtos/responses/auth"
```
again, why have requests and responses at the beginning here?

SemPlaatsman replied:
Cause else no worky :neckbeard:

Tayam-El-Jebari commented on docs/KEYCLOAK_SETUP_GUIDE.md:
whoops, will update this when automation script goes online

Tayam-El-Jebari commented on infrastructure/repositories/base_keycloak_repository.go:
```diff
@@ -0,0 +1,157 @@
+	return resp, nil
+}
+
+// getAdminToken retrieves an admin token from Keycloak
```
this comment is quite unnecessary, method speaks for itself.

SemPlaatsman replied:
Yesss tru

Tayam-El-Jebari commented on infrastructure/repositories/base_keycloak_repository.go:
```diff
@@ -0,0 +1,157 @@
+	return result.AccessToken, nil
+}
+
+// makeJSONRequest is a helper method for making HTTP requests with JSON body
```
same here

Tayam-El-Jebari commented on core/services/encryption_service.go:
You send the raw password to keycloak to handle  the password hashing internally in the identity repository. What is the reason we don't do that for the pin here?

SemPlaatsman replied:
Keycloak doesn't see the pin as a credential. Only a password and OTP are credentials.

Tayam-El-Jebari replied:
ripp

SemPlaatsman committed:
Added requested changes (see desc)

- Removed unnecessary comments
- Moved from `config.yaml` to `.env`
- Updated documentation
